Homework7/Hw7/MyHtmlServices/HtmlHelperExtensions.cs

```diff
@@ -7,6 +10,92 @@ public static class HtmlHelperExtensions
+        }
+
+        return resultingHtmlContent;
+    }
+    private static IHtmlContent CreateInputForProperty(PropertyInfo property, object? model)
+    {
+        if (property.PropertyType == typeof(string))
+        {
+            return CreateInputField(property.Name, "text", model == null ? null : property.GetValue(model) as string);
+        }
+        if (property.PropertyType == typeof(int))
+        {
+            return CreateInputField(property.Name, "number", model == null ? null : (property.GetValue(model) as int?).ToString());
+        }
+        return CreateDropdownFromEnum(property, model);
+    }
+
+    private static IHtmlContent CreateLabel(PropertyInfo property)
+    {
+        string? propertyDisplayName = null;
+        foreach (var attribute in property.GetCustomAttributes())
+        {
+            if (attribute is DisplayAttribute displayAttribute)
+                propertyDisplayName = displayAttribute.Name;
+        }
+
+        if (propertyDisplayName == null)
+        {
+            propertyDisplayName = FromCamelCaseToDisplayName(property.Name);
+        }
+        return new HtmlString($"<label for={property.Name}>{propertyDisplayName}</label>");
+    }
+
+    private static string FromCamelCaseToDisplayName(string propertyName)
+    {
+        return Regex.Replace(propertyName, "([A-Z])", " $1", RegexOptions.Compiled).Trim();
+    }
+
+    private static IHtmlContent CreateInputField(string name, string type, string? value)
+    {
+        return new HtmlString($"<input id=\"{name}\" type=\"{type}\" name=\"{name}\" value=\"{value}\"></input>");
+    }
+
+    private static IHtmlContent CreateDropdownFromEnum(PropertyInfo property, object? model)
+    {
+        var hcb = new HtmlContentBuilder();
+        hcb.AppendHtml($"<select id=\"{property.Name}\" name=\"{property.Name}\">");
+        foreach (var optionName in property.PropertyType.GetEnumNames()) 
+        {
+            bool isSelected = false;
+            if (model != null)
+            {
+                var value = property.GetValue(model);
+                isSelected = property.PropertyType.GetEnumName(value!) == (string)optionName;
+            }
+            hcb.AppendHtml($"<option value=\"{optionName}\" {(isSelected ? "selected" : string.Empty)}>{optionName}</option>");
+        }
+        hcb.AppendHtml("</select>");
+        return hcb;
+    }
+    
+    private static IHtmlContent CreateValidationField(PropertyInfo property, object? model)
+    {
+        var hcb = new HtmlContentBuilder();
+
+        hcb.AppendHtml($"<span id=\"{property.Name}\">");
+        if (model == null) 
+            return hcb.AppendHtml("</span>");
+        var attributes = property.GetCustomAttributes<ValidationAttribute>();
+        foreach (var validationAttribute in attributes)
+        {
+            if (validationAttribute.IsValid(property.GetValue(model))) continue;
+            hcb.Append(validationAttribute.ErrorMessage);
```

Comment:
```suggestion
            if (!validationAttribute.IsValid(property.GetValue(model)))
                 hcb.Append(validationAttribute.ErrorMessage);
```
Не стоит использовать continue когда можно сделать проще. Сложный control flow сложно верифицировать глазами